src/router_core/connections.c

```diff
@@ -1682,6 +1682,22 @@ static void qdr_connection_opened_CT(qdr_core_t *core, qdr_action_t *action, boo
         }
     } while (false);
 
+    const char *pname = conn->protocol_adaptor->name;
```

Comment:
I was wondering what is the impetus to track adaptor specific connections. A customer that is using skupper/skupper-router is ever only going to need to know how many connections were opened. They don't need to be grouped like this into separate connection counters. If you using the TCP adaptor, all you need to know how many connections were opened. Could we not get away with just adding a core->num_connections and just add and subtract from that when a qdr_connection is opened and closed ?

Also this could be confusing to users if we there were http connections being opened over tcp adaptor and the http1.x counter remains zero ?

Reply:
We already have a connectionCount attribute in the router management entity that counts all open connections (basically all active qdr_connection_t's). That count includes every type of connection: inter-router, management client, edge, tcp adaptor client, http/* adaptor client.

Note well that the proposed names are "<protocol>ServiceConnections"...

What we - or at least I - want is to be able to track the number of connections currently open for skupper *services*. That is, connections associated with provisioned "skupper service". And since a skupper service is provisioned for a particular protocol (via the --protocol parameter) we can further classify the connection to better understand the usage, which would be useful for deployments that combine different service protocol types.

Imagine being able to look into a prometheus log and see that an unexpected spike in HTTP/2 client connections happens every 24 hours a 2:30 pm. Or that the number of HTTP/1 clients stays flat but the TCP client connections keep accumulating. Or that the large number of allocated qdr_connection_t's corresponds to an unexpected peak load of HTTP/1 client request (not a leak).

That level of detail would be very useful.

> Also this could be confusing to users if we there were http connections being opened over tcp adaptor and the http1.x counter remains zero ?

They'd see tcpServiceConnections non-zero, which would make sense since they configured a TCP skupper service and not an HTTP skupper service. Why would they expect to see Http1ServiceConnections when they never configured an HTTP/1 service?

Reply:
> better understand the usage, which would be useful for deployments that combine different service protocol types.

My question was based on my understanding that customers (so far) have not mixed and match protocol adaptors. They have stayed with one protocol adaptor across the board. If we see that changing, this PR makes sense. I was just concerned that we are introducing something that might potentially confuse people although the `ServiceConnections` does make it clearer.

Reply:
It's my understanding that there's nothing preventing provisioning multiple services with different protocols. IIUC that's a fully supported feature of Skupper.

Is that not the case? If that is indeed true won't customers be confused if they provision TCP, HTTP/2 and HTTP/1 and only those client connections using one particular service is counted? That seems like a bug to me.